_posts/2018-12-24-snowplow-snowflake-loader-0.4.0-released/index.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: Snowplow Snowflake Loader 0.4.0 released
+title-short: Snowflake Loader 0.4.0
+tags: [snowflake, storage, relational databases]
+author: Rostyslav
+category: Releases
+permalink: /blog/2018/12/24/snowplow-snowflake-loader-0.4.0-released/
+---
+
+We are pleased to announce version 0.4.0 of the [Snowplow Snowflake Loader][snowflake-loader-repo]! This release introduces optional cross-batch natural deduplication and brings significant performance improvements to the Snowflake Transformer, and includes several other updates and bugfixes.
+
+Read on below the fold for:
+
+1. [Cross-batch deduplication](#crossbatch-dedupe)
+2. [S3 optimizations](#s3-optimizations)
+3. [New configuration options](#new-config)
+4. [Other changes](#other-changes)
+5. [Upgrading](#upgrading)
+6. [Getting help](#help)
+
+<!--more-->
+
+<h2 id="crossbatch-dedupe">1. Cross-batch deduplication</h2>
+
+It’s possible for 2 or more Snowplow events to have the same event ID due to the event ID being set in a Snowplow tracker rather than the pipeline. This can, in some cases, cause problems in Snowflake when an inner or left join becomes a cartesian product. Event duplicates can prove a challenge in any event pipeline - we have previously discussed this issue in detail in [this blog post][duplicate-blog] and on our [Discourse forum][duplicate-discourse].
```

Comment:
Sorry, missed the statement about cartesian product. Its not true generally for Snowflake, because we're not joining multiple tables unlike in Redsfhift.